### PR TITLE
Clone activity for terraform workflow

### PR DESCRIPTION
**Pull request by samrabelachew**
The clone activity will rely on both the GH client's GetArchiveLink() fxn to fetch the archive URL of a repo and hashicorp go-getter's GetAny() fxn to handle downloading, extracting, and filtering for necessary files.

I also went ahead and added the corresponding cleanup logic for the subdirectories we download.

Design decisions:
The GH client's archive link lasts 5 minutes. To reduce complexity, I wanted to keep both network calls (the initial archive link fetch and subsequent download) within the same `DownloadRepo` activity. If we split up those network calls into two activities, we would have to do more work storing/fetching the archive link elsewhere since it would contain a token, potential increasing time and opening us up to more points of failure that could delay the download request.

Using go-getter library just reduces our code complexity by letting us avoid dealing with generating temporary directories and migrating downloaded files to final destination.

### CODE REVIEW

File: server/neptune/workflows/internal/activities/github.go [hunk 1]
```diff
@@ -121,22 +127,41 @@ func (a *githubActivities) CreateCheckRun(ctx context.Context, request CreateChe
...
+	// note: this link exists for 5 minutes when fetching a private repository archive
+	archiveLink, resp, err := client.Repositories.GetArchiveLink(ctx, request.Repo.Owner, request.Repo.Name, github.Zipball, opts, true)
+	if err != nil {
+		return DownloadRootResponse{}, errors.Wrap(err, "getting repo archive link")
+	}
+	// GH responds with a 302 + redirect link to where the archive exists
+	if resp.StatusCode != http.StatusFound {
+		return DownloadRootResponse{}, errors.Errorf("getting repo archive link returns non-302 status %d", resp.StatusCode)
+	}
+	downloadLink := a.LinkBuilder.BuildDownloadLinkFromArchive(archiveLink, request.Root, request.Repo)
```
Comment (samrabelachew) [inline server/neptune/workflows/internal/activities/github.go:158]: I split this out to perform some extra testing, but is the interface excessive? I can just have it act as a helper fxn without the interface/empty struct.

File: server/neptune/workflows/internal/activities/github.go [hunk 2]
```diff
@@ -121,22 +128,44 @@ func (a *githubActivities) CreateCheckRun(ctx context.Context, request CreateChe
...
+	DeploymentId string
 }
 
-type GithubRepoCloneResponse struct {
+type DownloadRootResponse struct {
 	LocalRoot *root.LocalRoot
 }
 
-func (a *githubActivities) GithubRepoClone(ctx context.Context, request GithubRepoCloneRequest) (*GithubRepoCloneResponse, error) {
+func (a *githubActivities) DownloadRoot(ctx context.Context, request DownloadRootRequest) (DownloadRootResponse, error) {
```
Comment (nishkrishnan) [inline server/neptune/workflows/internal/activities/github.go:141]: Mind adding a comment on the process of what's going on in this function?

Comment (nishkrishnan) [inline server/neptune/workflows/internal/activities/github.go:141]: no unit tests for this?

File: server/neptune/workflows/internal/github/link/builder.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+// BuildDownloadLinkFromArchive is a helper fxn that isolates the logic of modifying a GH archive link
+// into source url that the go-getter library understand for downloading
+func (b Builder) BuildDownloadLinkFromArchive(archiveURL *url.URL, root root.Root, repo github.Repo) string {
+	// Add archive query parameter for getter library to extract archive
+	queryParams := "archive=zip"
+	token := archiveURL.Query().Get("token")
+	if token != "" {
+		queryParams += fmt.Sprintf("&token=%s", token)
+	}
```
Comment (nishkrishnan) [inline server/neptune/workflows/internal/github/link/builder.go:21]: when would this be empty or not empty?

File: server/neptune/workflows/internal/activities/github.go [hunk 4]
```diff
@@ -121,22 +128,44 @@ func (a *githubActivities) CreateCheckRun(ctx context.Context, request CreateChe
...
-	DestinationDir string
-	Root           root.Root
+type DownloadRootRequest struct {
+	Repo         internal.Repo
+	Root         root.Root
+	DeploymentId string
 }
 
-type GithubRepoCloneResponse struct {
+type DownloadRootResponse struct {
```
Comment (nishkrishnan) [inline server/neptune/workflows/internal/activities/github.go:137]: I would just called it FetchRoot or GetRoot or ResolveRoot cuz this is doing more than downloading

**Review by nishkrishnan — COMMENTED** [review]
Looks good, only minor stuff

File: server/neptune/workflows/internal/github/link/builder.go (hunk 3 above)
Comment (samrabelachew) [reply server/neptune/workflows/internal/github/link/builder.go:21]: Public vs. private repository I believe (at least from some testing). The GH docs are pretty vague on this: https://docs.github.com/en/rest/repos/contents#download-a-repository-archive-zip

**Review by nishkrishnan — APPROVED** [review]
